### ISSUE TITLE
Error messages are truncated

### ISSUE THREAD

**Issue by HaleTom** [zdharma/zplugin#59]
**Describe the bug**
When loading `zsh`, some error messages are truncated.

**To Reproduce**
`.zshrc`:
```
alias -g C='| wc -l'
source "$XDG_DATA_HOME/zplugin/bin/zplugin.zsh"
autoload -Uz _zplugin
(( ${+_comps} )) && _comps[zplugin]=_zplugin
zplugin ice wait'0' pick'zsh-vim-mode.plugin.zsh'
zplugin light softmoth/zsh-vim-mode
```
1. Start a new shell and see a parse error message.
1. Then start yet another new shell.

The cause of the issue is the the global alias `alias -g C='| wc -l'` causing a syntax error.  This is a specific illustration of a general problem. See *Additional context* for more.

**Expected behavior**
I expect that the line shown below (`/home/ra` == part of my home directory name) is not truncated, and if there are more than one error lines, that they  are fully shown.

**Screenshots**
On loading a new shell, I see:
```
svelte% 
/home/ra
Loaded softmoth/zsh-vim-mode
```

**Versions:**
 - $ZSH_VERSION == 5.5.1
 - Operating system name and version (uname -a): 
```
Linux svelte 4.16.18-1-MANJARO #1 SMP PREEMPT Tue Jun 26 15:27:59 UTC 2018 x86_64 GNU/Linux
```

**Additional context**
I will work on a PR to disable aliases at `zcompile`-time separately. This issue about error messages being truncated generally, not the cause of this specific error message.

**Comment by HaleTom** [comment]
The background of why the syntax error is happening with `softmoth/zsh-vim-mode` can be found [here](https://github.com/softmoth/zsh-vim-mode/issues/3).

**Comment by HaleTom** [comment]
Thank you! :)

**Comment by HaleTom** [comment]
Sorry, I must have made a config change which hid this issue - it still persists.

Here's a minimal `.zshrc` which illustrates it:

```
source "$XDG_DATA_HOME/zplugin/bin/zplugin.zsh"
autoload -Uz _zplugin
# shellcheck disable=SC2154,2034,2004
(( ${+_comps} )) && _comps[zplugin]=_zplugin

alias -g C='| wc -l'
zplugin ice wait'0'
zplugin light 'softmoth/zsh-vim-mode'
return
```

The output after running `zsh` is:

```
svelte%
/home/ra
Loaded softmoth/zsh-vim-mode
```

Related: PR Prevent parse errors caused by aliases #64.

Would you please reopen this?

**Comment by psprint** [comment]
I have an obstacle, the global alias doesn't result in error for me. Thinking what to do next.

**Comment by HaleTom** [comment]
Here's a `.zshrc` with no dependencies which should illustrate the problem:

```
declare -A ZPLGM
ZPLGM[HOME_DIR]=$XDG_DATA_HOME/zplugin
source "$XDG_DATA_HOME/zplugin/bin/zplugin.zsh"
autoload -Uz _zplugin
(( ${+_comps} )) && _comps[zplugin]=_zplugin

dir=$ZPLGM[PLUGINS_DIR]/borken
mkdir -p -- $dir
echo if >| $dir/init.sh
zplugin ice wait'0'
zplugin load borken

return
```

It gives:
```
svelte% exec zsh
svelte%
/home/ra
Loaded borken
```

Note that because the line with the syntax error is truncated to the length of the prompt, it makes hunting down the syntax error quite hard.

The work-around is to set an extremely long prompt :)

**Comment by psprint** [comment]
I will recheck my previous method with `.reset-prompt`, maybe it has some glitch but overall works. That said, I've added other mechanism: return codes of `source`s are remembered, summed up, returned, not to `zplugin load ...`, etc. (just need to think that through, if it's good to return e.g. 1 from `zplugin load ...` if script failes to parse or does `return 1`), but to function displaying `Loaded ...` message – yes.

In short: error codes flow from `source` to upper functions and `Loaded ...` message is omitted when error code != 0. This stops message truncation.

**Comment by psprint** [comment]
I've finally checked on the previous method. There was indeed a glitch.

The previous method doesn't help in the last provided case (`... echo if >| $dir/init.sh ...`), but should help when the plugin is being downloaded first. So there are 2 methods that should help to see the error message without truncation.

**Comment by HaleTom** [comment]
Thanks a bunch!
